src/snowflake/snowpark/mock/connection.py

```diff
@@ -405,8 +410,27 @@ def execute(
         elif isinstance(res, list):
             rows = res
 
+        if to_pandas:
+            pandas_df = pd.DataFrame()
+            for col_name in res.columns:
+                pandas_df[unquote_if_quoted(col_name)] = res[col_name].tolist()
+            rows = _fix_pandas_df_integer(res)
+
+            # the following implementation is just to make DataFrame.to_pandas_batches API workable
+            # in snowflake, large data result are split into multiple data chunks
+            # and sent back to the client, thus it makes sense to have the generator
+            # however, local testing is designed for local testing
+            # we do not mock the splitting into data chunks behavior
+            if to_iter:
```

Comment:
Why can't we reuse the if block on line 431?

Reply:
document the generator vs iter behavior upon PD frame